### PR TITLE
Add recipe for pyDOE

### PR DESCRIPTION
**Pull request by melund**
pyDOE is a package for 'Design of experiments'. It is an optional dependency in one of my own libraries so I will be happy to maintain this recipe. 

Snippet from https://pythonhosted.org/pyDOE/ :
The pyDOE package is designed to help the scientist, engineer, statistician, etc., to construct appropriate experimental designs.

### CODE REVIEW

**Comment by patricksnape** [comment]
Sorry, there was a problem with the CI server and so hachoir-core and hyperspy were not removed from the repo. Could you please rebase this PR on top of master (I just fixed it)?

**Comment by melund** [comment]
@patricksnape .. I have rebased.

File: recipes/pydoe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - pyDOE
+
+about:
+  home: https://github.com/tisimst/pyDOE
+  license: BSD License (3-clause)
+  summary: The pyDOE package is designed to help the scientist, engineer, statistician, etc., to construct appropriate experimental designs.
+
+extra:
+  recipe-maintainers:
+    - melund
```
Comment (jakirkham) [inline recipes/pydoe/meta.yaml]: nit: please add a terminal newline.

File: recipes/pydoe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+build:
+  script: python setup.py install
+  number: 0
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - scipy
```
Comment (jakirkham) [inline recipes/pydoe/meta.yaml]: Is `scipy` needed as a build requirement or can it be dropped?

File: recipes/pydoe/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  name: pydoe
+  version: {{ version }}
+
+source:
+    fn: pyDOE-afc5949f01c010a26764adba9624871d51797270.tar.gz
+    url: https://github.com/tisimst/pyDOE/archive/afc5949f01c010a26764adba9624871d51797270.tar.gz
+    sha1: 54f87d106c3f49e1b4cff17a31ef8a9e0142ec5b
+
+build:
+  script: python setup.py install
```
Comment (jakirkham) [inline recipes/pydoe/meta.yaml]: Please add the arguments `--single-version-externally-managed --record=record.txt` when using `setuptools`.

File: recipes/pydoe/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    - scipy >=0.13
+    - numpy >=1.8
+
+test:
+  imports:
+    - pyDOE
+
+about:
+  home: https://github.com/tisimst/pyDOE
+  license: BSD License (3-clause)
```
Comment (jakirkham) [inline recipes/pydoe/meta.yaml]: Personally I prefer `BSD 3-Clause`.

File: recipes/pydoe/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{% set version = "0.3.8" %}
+
+package:
+  name: pydoe
+  version: {{ version }}
+
+source:
+    fn: pyDOE-afc5949f01c010a26764adba9624871d51797270.tar.gz
+    url: https://github.com/tisimst/pyDOE/archive/afc5949f01c010a26764adba9624871d51797270.tar.gz
```
Comment (jakirkham) [inline recipes/pydoe/meta.yaml]: Could we please ask them to add a git tag?

File: recipes/pydoe/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    - numpy >=1.8
+
+test:
+  imports:
+    - pyDOE
+
+about:
+  home: https://github.com/tisimst/pyDOE
+  license: BSD License (3-clause)
+  summary: The pyDOE package is designed to help the scientist, engineer, statistician, etc., to construct appropriate experimental designs.
```
Comment (jakirkham) [inline recipes/pydoe/meta.yaml]: Might be good if we could shorten this a bit. A longer form is ok as a `description`.

File: recipes/pydoe/meta.yaml (hunk 2 above)
Comment (melund) [inline recipes/pydoe/meta.yaml]: Probably not really needed, but the setup.py has `install_requires=['numpy', 'scipy'],` so my outdated experience with conda-build tells me that it will complain otherwise. Anyway I will just try.

File: recipes/pydoe/meta.yaml (hunk 5 above)
Comment (melund) [inline recipes/pydoe/meta.yaml]: Yes. I can ask. Is this blocking? I can update the recipe if they add a git-tag later.

File: recipes/pydoe/meta.yaml (hunk 2 above)
Comment (melund) [inline recipes/pydoe/meta.yaml]: Well. Seems to work without. So I will just remove.

**Comment by melund** [comment]
@jakirkham Thanks for the comments. I have update the recipe.

File: recipes/pydoe/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+test:
+  imports:
+    - pyDOE
+
+about:
+  home: https://github.com/tisimst/pyDOE
+  license: BSD 3-Clause
+  summary: Design of experiments for Python
+  description: The pyDOE package is designed to help the scientist, engineer, statistician, etc., to construct appropriate experimental designs.
```
Comment (patricksnape) [inline recipes/pydoe/meta.yaml]: As noted by @jakirkham, would be good to shorten this to something like `Package for constructing experimental designs`

Comment (melund) [inline recipes/pydoe/meta.yaml]: This is what I did. The longer version is moved to the description section. It is OK for the description to be more verbose. This the field that GUI tools use to provide the user with a more detailed description of what the package does.

 The actual sentences are those of the author from the docs.

Comment (patricksnape) [inline recipes/pydoe/meta.yaml]: Yup! My bad 😆

**Comment by patricksnape** [comment]
@jakirkham This looks ready bar the lack of a tag? Should we merge anyway?

File: recipes/pydoe/meta.yaml (hunk 5 above)
Comment (jakirkham) [inline recipes/pydoe/meta.yaml]: xref: https://github.com/tisimst/pyDOE/issues/5

Comment (jakirkham) [inline recipes/pydoe/meta.yaml]: Oh, they have it on [PyPI]( https://pypi.python.org/pypi/pyDOE/0.3.8 ). Could we switch to that? We can just use this as the url ( `https://pypi.io/packages/source/p/pyDOE/pyDOE-{{ version }}.zip` ).

**Comment by jakirkham** [comment]
The tag is kind of important. I want to make sure we are not defining the release for upstream. Though they do add the version in a commit. However it could mean it is a dev release.

That being said, it looks like we can use PyPI, which does have the 0.3.8 release. Switching to that would make this ready to go IMHO.

File: recipes/pydoe/meta.yaml (hunk 5 above)
Comment (melund) [inline recipes/pydoe/meta.yaml]: Nice. I didn't know you could still get nice looking links to pypi.

**Comment by melund** [comment]
@conda-forge/core  I updated with the PyPi link as @jakirkham suggested.b
